src/common/utils.hh

```diff
@@ -53,6 +53,25 @@ inline uint32_t leadingSpaceCount( const QString & str )
   return 0;
 }
 
+inline QString trimQuotes( QString const & str ) const
+{
+  QString trimmed( str );
+
+  int n = 0;
+  while ( n < str.length() && ( str[ n ] == '\"' || str[ n ] == '\'' ) ) {
+    n++;
+  }
+  if ( n > 0 ) {
+    trimmed = trimmed.mid( n );
+  }
+
+  while ( trimmed.endsWith( '\"' ) || trimmed.endsWith( '\'' ) ) {
+    trimmed.chop( 1 );
+  }
+
+  return trimmed;
```

Comment:
```suggestion
  const auto* begin = str.cbegin();
  const auto* end = str.cend();
  while (begin < end && end[-1] == '\"')
  {
    --end;
  }
  while (begin < end && *begin == '\"')
  {
    begin++;
  }
  return str.sliced(begin - str.cbegin(), end - begin);
```

Reply:
I think it is the same . just two different style.

Reply:
mid is returning a new string replacing the old one.
https://doc.qt.io/qt-6/qstring.html#mid

One single `sliced` is one allocation and no more.

Reply:
slice was a reference to the original string which most likely will cause crash if the original string is destroyed.

Reply:
https://doc.qt.io/qt-6/qstring.html#sliced

sliced return a copy.

Reply:
Have you tested it ?  end[-1] or `*end[-1]`

Reply:
I'll stick with the previous one ,more readable.

Reply:
end[-1] is equivalent to *(end - 1).

So end [-1]

Reply:
https://github.com/qt/qtbase/blob/dev/src%2Fcorelib%2Ftext%2Fqstringalgorithms_p.h#L57-L73

Reply:
<img width="300" src="https://github.com/user-attachments/assets/c5478985-78da-4c82-9e7f-f69278728885" />